### DISCUSSION TITLE
What's the future of jQuery?

### DISCUSSION THREAD

**Discussion by kenprogrammer** [jquery/jquery#4907] [Q&A] [answered]
Since bootstrap dropped jQuery & rumour has it that ES6 can now do what jQuery does. Cross-browsers compatibility is no longer an issue. What's the future of this library? Should I still use it in brand new projects?

**Comment by timmywil** [comment, marked as answer]
Thanks for opening a discussion. Plans for the future remain largely the same as they were in [this comment](https://github.com/jquery/jquery/issues/3886#issuecomment-350880747). We will continue to support jQuery as long as it is used, and usage is still very high globally. While jQuery 4 has been delayed and taken longer than expected, we have released several 3.x versions since then and mostly stuck to our goal of 2 versions per year.

There are few things about your statements that aren't quite correct. First, bootstrap didn't drop jQuery exactly. They created jQuery clone that they once called "jQuery lite" with only the features that they needed, but if jQuery is included anywhere on the page, bootstrap still uses jQuery. And the version of EcmaScript that was once called es6 did not have all jQuery features, nor do current web APIs or the current version of EcmaScript. That said, the web has a lot more these days and many features that have made it to the web were inspired by jQuery. Also, cross-browser compatibility is still an issue and the jQuery maintainers still have to deal with many differences in behavior between browsers. Of course, it's not nearly as bad as it was in the old days. But the most attractive thing to me about jQuery was never any of these things. It was the API.

All that said, I will tell you that I personally no longer use jQuery in projects, but most of my projects are large, complex web applications that often work at a higher level. I might still use jQuery in smaller web sites, especially if I were to find myself writing my own DOM utilities. Point is, the decision about whether to use jQuery in a project could generate a much longer discussion and is more nuanced than you might expect. We no longer evangelize jQuery usage or have jQuery conferences and such, but we also do not discourage its usage. Regardless of that discussion, though, jQuery needs to be supported for all the users it does have and we will continue to do that.

Thanks.

**Reply by kenprogrammer** [reply]
Thanks for the clarification.
